test/Table_Tests/src/Database/Postgres_Spec.enso

```diff
@@ -316,13 +316,13 @@ postgres_specific_spec connection db_name setup =
             col.replace patterns replacements case_sensitivity=Case_Sensitivity.Sensitive . to_vector . should_equal ['bye Hello', 'bye bye', 'HELLO HELLO', 'hey A[bCd] hey', 'abac ad Ab aCAd']
 
         Test.specify "case_sensitivity=sensitive/default use_regex=false only_first=true"
-            col.replace patterns replacements only_first=True . should_fail_with Unsupported_Database_Operation
+            col.replace patterns replacements only_first=True . to_vector . should_fail_with Unsupported_Database_Operation
```

Comment:
It does not feel right that `Unsupported_Database_Operation` is only reported _after materializing_ the result.

I think we need to try checking this at the time the query is built. That's one advantage of the old approach, although still the new one feels cleaner to me. Maybe we can define `is_replace_setting_supported` to `Dialect`, to check the support at build-time?

Reply:
I've implemented are_replace_params_supported; I was hoping to have a map from supported parameter values to implementation functions, but it just wasn't practical, so I have a separate list of supported combinations, which has to be kept in sync with the implementation of the op, but I think that's the best we can do.

Reply:
I think that as long as we test _both_ supported and unsupported paths (by checking 'unsupported' error being reported), we are good - the unit tests will then be ensuring that these are synchronized.